src/phoenix/server/api/dataloaders/span_dataset_examples.py

```diff
@@ -0,0 +1,35 @@
+from typing import (
+    AsyncContextManager,
+    Callable,
+    List,
+)
+
+from sqlalchemy import select
+from sqlalchemy.ext.asyncio import AsyncSession
+from strawberry.dataloader import DataLoader
+from typing_extensions import TypeAlias
+
+from phoenix.db import models
+
+SpanID: TypeAlias = int
+Key: TypeAlias = SpanID
+Result: TypeAlias = List[models.DatasetExample]
+
+
+class SpanDatasetExamplesDataLoader(DataLoader[Key, Result]):
+    def __init__(self, db: Callable[[], AsyncContextManager[AsyncSession]]) -> None:
+        super().__init__(load_fn=self._load_fn)
+        self._db = db
+
+    async def _load_fn(self, keys: List[Key]) -> List[Result]:
+        span_ids = keys
```

Comment:
We've been using `rowid` for disambiguation (so the reader doesn't need a double take) since `span_id` is usually thought of as `str`.
```suggestion
        span_rowids = keys
```

Reply:
oh gotcha, makes sense!